pkg/alertmanager/statefulset.go

```diff
@@ -117,6 +117,11 @@ func makeStatefulSetSpec(a *spec.Alertmanager) v1beta1.StatefulSetSpec {
 		fmt.Sprintf("-mesh.listen-address=:%d", 6783),
 		fmt.Sprintf("-storage.path=%s", "/etc/alertmanager/data"),
 	}
+
+	if a.Spec.ExternalUrl != "" {
+		commands = append(commands, "-web.external-url="+a.Spec.ExternalUrl)
```

Comment:
This is a sequence of commands as they are represented after the arguments have been parsed initially, so quoting results in the quotes being an actual part of the flags string value:

When trying this in the Prometheus StatefulSet:

```
level=error msg="invalid Prometheus URL: \"http://192.168.99.100:30900/prometheus\""
```

When trying this in the Alertmanager StatefulSet:

```
panic: wildcards must be named with a non-empty name in path '/"http://192.168.99.100:30903/alertmanager"/app/*filepath'
```